fastpair/rust/src/bluetooth/windows/device.rs

```diff
@@ -49,6 +74,97 @@ impl Device for BleDevice {
     fn name(&self) -> Result<String, anyhow::Error> {
         Ok(self.inner.Name()?.to_string_lossy())
     }
+
+    fn address(&self) -> Address {
+        Address::Ble(self.addr)
+    }
+
+    async fn pair(&self) -> Result<(), anyhow::Error> {
+        // BLE Audio isn't supported on Windows natively, so devices can pair
+        // but don't playback. Might possibly work with UWP. Since the Classic
+        // and BLE APIs are very similar, it might be possible to copy-paste
+        // `ClassicDevice::pair` directly.
+        unimplemented!("BLE Pairing is currently unsupported.")
+    }
+}
+
+
+impl ClassicDevice {
+    /// `ClassicDevice` constructor.
+    pub async fn new(addr: ClassicAddress) -> Result<Self, anyhow::Error> {
+        let raw_addr = u64::from(addr);
+
+        let inner = BluetoothDevice::FromBluetoothAddressAsync(
+            raw_addr,
+        )?
+        .await?;
+
+        Ok(ClassicDevice { inner, addr })
+    }
+}
+
+#[async_trait]
+impl Device for ClassicDevice {
+    fn name(&self) -> Result<String, anyhow::Error> {
+        Ok(self.inner.Name()?.to_string_lossy())
```

Comment:
can we use normal to_string() here? i feel like windows requires the adapter names to be utf-8 encoded.